src/business-logic/services/database-change-ingest-service.ts

```diff
@@ -40,46 +55,170 @@ export class DatabaseChangeIngestService implements IngestChangeService {
-    rundownChangeListener.onDeleted(rundownId => this.enqueueEvent(() => this.deleteRundown(rundownId)))
+  private listenForPartChanges(partChangedListener: DataChangedListener<IngestedPart>): void {
+    partChangedListener.onCreated(part => this.enqueueEvent(3, () => this.createPart(part)))
+    partChangedListener.onUpdated(part => this.enqueueEvent(4, () => this.updatePart(part)))
+    partChangedListener.onDeleted(partId => this.enqueueEvent(7, () => this.deletePart(partId)))
   }
 
-  private listenForSegmentChanges(segmentChangedListener: DataChangedListener<Segment>): void {
-    segmentChangedListener.onCreated(segment => this.enqueueEvent(() => this.createSegment(segment)))
-    segmentChangedListener.onUpdated(segment => this.enqueueEvent(() => this.updatedSegment(segment)))
-    segmentChangedListener.onDeleted(segmentId => this.enqueueEvent(() => this.deleteSegment(segmentId)))
+  private async syncDataWithIngestedData(): Promise<void> {
+    const ingestedRundowns: IngestedRundown[] = await this.ingestedRundownRepository.getIngestedRundowns()
+    await this.deleteRundownsNotPresentInIngestedRundowns(ingestedRundowns)
+
+    await Promise.all(ingestedRundowns.map(async (ingestedRundown) => {
+      let oldRundown: Rundown | undefined
+      try {
+        oldRundown = await this.rundownRepository.getRundown(ingestedRundown.id)
+      } catch (exception) {
+        if (!(exception instanceof NotFoundException)) {
+          throw exception
+        }
+        // The Rundown doesn't exist in the database which means it's a brand new Rundown.
+      }
+
+      // If the Rundown isn't active, we can simply just "re-ingest" it into our database collection as a fresh Rundown.
+      const updatedRundown: Rundown = oldRundown && oldRundown.isActive()
+        ? this.updateActiveRundownFromIngestedRundown(ingestedRundown, oldRundown)
+        : this.createNewRundownFromIngestedRundown(ingestedRundown)
+
+      await this.rundownRepository.deleteRundown(ingestedRundown.id) // Delete the old Rundown to get rid of deleted Entities
+      await this.rundownRepository.saveRundown(updatedRundown) // Save the new Rundown
+      this.eventEmitter.emitRundownUpdated(updatedRundown)
+    }))
+  }
+
+  private async deleteRundownsNotPresentInIngestedRundowns(ingestedRundowns: IngestedRundown[]): Promise<void> {
+    const ingestedRundownsIds: string[] = ingestedRundowns.map(ingestedRundown => ingestedRundown.id)
+    const basicRundowns: BasicRundown[] = await this.rundownRepository.getBasicRundowns()
+    const rundownIds: string[] = basicRundowns.map(basicRundown => basicRundown.id)
+
+    for (const rundownId of rundownIds) {
+      // Delete all Rundowns that has been deleted and therefore no longer exists in the IngestedRundowns.
+      if (!ingestedRundownsIds.includes(rundownId)) {
+        await this.rundownRepository.deleteRundown(rundownId)
+        this.eventEmitter.emitRundownDeleted(rundownId)
+      }
+    }
+  }
```

Comment:
Mapping basic rundowns to ids seems unnecessary. Here is a suggestion:
```typescript
  private async deleteRundownsNotPresentInIngestedRundowns(ingestedRundowns: IngestedRundown[]): Promise<void> {
    const ingestedRundownsIds: string[] = ingestedRundowns.map(ingestedRundown => ingestedRundown.id)
    const basicRundowns: BasicRundown[] = await this.rundownRepository.getBasicRundowns()
    
    for (const basicRundown of basicRundowns) {
      if (ingestedRundownsIds.includes(basicRundown.id) {
        continue
      }
      await this.rundownRepository.deleteRundown(basicRundown.id)
      this.eventEmitter.emitRundownDeleted(basicRundown.id)
    }
  }
```

I have removed the comment in the suggestion, since it says the same as the method name.

Reply:
Updated